rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -44,33 +61,87 @@
  */
 public class FlowNodeUtil {
 
+    protected static final Cache<String,List<FlowNode>> executionCache = CacheBuilder.newBuilder().weakValues().maximumSize(100).build();
```

Comment:
Probably a :bug: since there is not likely anyone holding this `List` strongly, so the cache will get discarded on every GC cycle.

When I said you should avoid holding `FlowNode`s strongly in a cache, what I meant is that you should change the type of the cache, to their `String` IDs for example.